how-to-guides/sequential-pipelines/scripts/data_preprocessing.py

```diff
@@ -0,0 +1,42 @@
+import neptune.new as neptune
+from sklearn.datasets import fetch_lfw_people
+from utils import *
+
+# Download dataset
+dataset = fetch_lfw_people(min_faces_per_person=70, resize=0.4)
+
+# (Neptune) Create a new run
+run = neptune.init_run(
+    monitoring_namespace="monitoring/preprocessing",
+)
+
+# Get dataset details
+dataset_config = {
+    "target_names": str(dataset.target_names.tolist()),
+    "n_classes": dataset.target_names.shape[0],
+    "n_samples": dataset.images.shape[0],
+    "height": dataset.images.shape[1],
+    "width": dataset.images.shape[2],
+}
+
+# (Neptune) Set basenamespace
+handler_run = run["preprocessing"]
```

Comment:
I'm a bit concerned that it may be tough to grasp what happens here if you're not familiar with Neptune terminology.

- The comment above could be a bit more detailed/helpful. Maybe something like:

    `# (Neptune) Set up "preprocessing" namespace inside the run. This will be the base namespace where all the preprocessing metadata is logged.`

- Might there be a more user-friendly name for this namespace handler object? Something more use-case oriented which expresses that we're dealing with a particular step that represents a part of the run.

    Given `run`, then maybe something like `run_preprocessing`? Ideas are welcome.

Reply:
I absolutely agree that the comment can be improved, thanks for sharing! I'll update it.

In regards to the second part, I'm not entirely sold on it but don't have super strong opinions.

Because when I look at it, it's actually a handler object and not a run. We must educate them on our terminologies (i.e., explain what a handler is). Just like we speak of `run` instead of `experiment`.

Reply:
Exactly -- I would like to try avoiding a variable name that may lead users to think this object is some kind of run in itself. I still have concerns when it comes to `handler_run`, as I believe it's more accurate to say it's a _namespace_ handler.

From a language perspective, it's tricky to succinctly express "part of a run". But I agree we need not totally avoid the term "handler" since we do use that in our docs. We've had this discussion within the product team, in which we agreed that we should try more "use-case oriented" handler names if feasible in the given context.

But perhaps with the more helpful comment, we could use `namespace_handler` here? Would it be too verbose to prefix the specific step (`preprocessing_namespace_handler`, `training_namespace_handler`)? That might make things even more clear.

Reply:
Or even just `preprocessing_namespace`, `training_namespace`, etc.

Reply:
We could do **`preprocessing_handler`**. At least the user can use the keyword `handler` in the naming if they want to learn more about the object.

Reply:
updated 👍